arch/xtensa/include/kernel_arch_func.h

```diff
@@ -122,7 +122,7 @@ static inline void _IntLibInit(void)
 }
 #endif
 
-#define _is_in_isr() (_arch_curr_cpu()->nested != 0)
+#define _is_in_isr() (_arch_curr_cpu()->nested != 0U)
```

Comment:
Not really understanding this pattern.  Is the checking tool really flagging these?  The literals "0" and "0u" are equivalent after promotion per the standard.  I mean, I buy that you want signedness to match in the general case, but SURELY the MISRA stuff isn't demanding that it's impossible to compare "int" with "0", right?

Reply:
The standard doesn't let anything be implicit. We need to explicitly state and make it clear that the constant is unsigned int.